### DISCUSSION TITLE
zh/katex

### DISCUSSION THREAD

**Discussion by giscus** [mdit-plugins/mdit-plugins#47] [General]
# zh/katex

使用 KaTeX 呈现数学表达式的插件，你应该将 katex 安装为 peer 依赖。 注 这个插件基于 。 使用 我们也有一个 @mdit/plugin-katex-slim 包，其中 katex 是可选依赖。 格式 你应该在行内使用 $tex expression$，在块中使用 $tex expression$。 样式 你应该自己从 katex...

https://mdit-plugins.github.io/zh/katex.html

**Comment by zkeq** [comment]
Why does it throw an error following the documentation example?

```js
TypeError: (0 , import_node_module.createRequire) is not a function
    at plugin.ts:15:17
```

I encountered this error when using this plugin, as well as another one called mathjax. How can I fix it?

![image](https://github.com/user-attachments/assets/0ae1fe07-c29f-4525-a1b1-42b5f331d2d6)

Here is my code:

```ts
import MarkdownIt from 'markdown-it'
import { katex } from "@mdit/plugin-katex";

const mdi = new MarkdownIt()
mdi.use(katex); // This line causes the error

mdi.render(value)
```

Complete reproduction code: https://github.com/zkeq/markdown-latex/tree/katex

https://github.com/zkeq/markdown-latex/blob/bfb75cecd507d2ce056f450ece72ab75b4865aa2/src/App.vue#L2-L23

**Reply by Mister-Hope** [reply]
This should be fixed, fell free to open a new issue if the issue exisits

**Comment by xarthurx** [comment]
same problem here.

**Reply by Mister-Hope** [reply]
This should be fixed, fell free to open a new issue if the issue exisits

**Comment by xarthurx** [comment]
When using the code;
```
if (import.meta.client) {
	renderer.use(katex);
        console.log("using client side rendering.")
	}
else{
    console.log("using SSR>")
  }

```

I realize that `nuxt3` is using my `markdownit` plugin on the server-side, and probably that's the reason why `katex` doesn't work...

Is there a way to integrate `katex` in the server side?

I was using normal `markdown-it-katex` and it seems everything was fine.

**Comment by Leovenn** [comment]
是否可以支持 \\[ 或者 \\( 作为分隔符

**Reply by Mister-Hope** [reply]
Supported

**Comment by zhangbin110** [comment]


**Reply by zhangbin110** [reply]
不小心发了测试数据出去，申请删除
